src/ResultSetBuilder.php

```diff
@@ -27,51 +30,52 @@ public function __construct(Client $client)
     }
 
     public function buildResultSet(Response $response, Query $query): ResultSet
-    {
-        $results = $this->buildResults($response);
-
-        return new ResultSet($response, $query, $results);
-    }
-
-    public function buildModelFromIndexAndData($indexName, $data)
-    {
-        $pureIndexName = $this->client->getPureIndexName($indexName);
-        $indexToClass = $this->client->getConfig(Client::CONFIG_INDEX_CLASS_MAPPING);
-        if (!isset($indexToClass[$pureIndexName])) {
-            throw new RuntimeException(sprintf('Unknown class for index "%s", did you forgot to configure "%s"?', $pureIndexName, Client::CONFIG_INDEX_CLASS_MAPPING));
-        }
-
-        $context = $this->client->getSerializerContext($indexToClass[$pureIndexName]);
-
-        return $this->client->getSerializer()->denormalize($data, $indexToClass[$pureIndexName], null, $context);
-    }
-
-    private function buildResults(Response $response): array
     {
         $data = $response->getData();
         $results = [];
 
         if (!isset($data['hits']['hits'])) {
-            return $results;
+            return new ResultSet($response, $query, []);
         }
```

Comment:
Not sure this should be changed as it will break the compatibility with Elastica: https://github.com/ruflin/Elastica/blob/cbc962e22060cf7b1bafd403cf3027943e20ae15/src/ResultSet/DefaultBuilder.php#L32-L34

Reply:
It's the same behavior than before :) 
I inline the method, that's why there is this diff 🌻

Reply:
No, before it returned an array, now it return a ResultSet.

Reply:
Double no :)

Reply:
ok I see, my bad.